Reloading resources from "Debug" menu

Vankata453 opened the pull request:
The new "Reload Resources" option in the "Debug" menu allows for reloading all fonts, textures, sprites and tilesets.

**Additionally:**

* Removes the `name` field for sprites. I honestly didn't know it existed before, and don't know what the point of it is, considering it's not referenced anywhere in the code, other than in logs, where a filename can be used instead.

Closes #2901.

Frostwithasideofsalt reviewed (approved):
Thank the gato gods 🙏🏼

tobbi commented on src/supertux/tile_set.cpp:
```diff
@@ -55,6 +56,18 @@ TileSet::TileSet() :
+{
+  m_autotilesets.clear();
+  m_thunderstorm_tiles.clear();
+  m_tiles.resize(1);
```
Why `1` in particular?

Vankata453 replied:
To preserve the initial tile with an ID of 0. I can add a comment for this.

tobbi commented on src/video/sdl_surface_ptr.hpp:
```diff
@@ -81,6 +81,12 @@ class SDLSurfacePtr final
     SDL_FreeSurface(m_surface);
     m_surface = surface;
   }
+  void reset(SDLSurfacePtr& other)
```
Add a new line before this.

Vankata453 replied:
To preserve the initial tile with an ID of 0. I can add a comment for this.

Vankata453 replied:
Is it needed, since it's an overload of the same function?

tobbi replied:
I'd still think so. Those things irk me.

tylerandari13 commented:
My train of thought to test this was to load the game, and remove an image. If it works then it should display a missing texture, because the images are no longer there, which resulted in this:
```
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:84 Image '/res/industrial/gfx-enemy/duker/duker-body.png' doesn't exist. Attempting to load ".deprecated" version.
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:439 Couldn't load texture '/res/industrial/gfx-enemy/duker/duker-body.png' (now using dummy texture): Couldn't open '/res/industrial/gfx-enemy/duker/duker-body.deprecated.png': not found
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:84 Image '/res/industrial/gfx-enemy/duker/duker-end.png' doesn't exist. Attempting to load ".deprecated" version.
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:439 Couldn't load texture '/res/industrial/gfx-enemy/duker/duker-end.png' (now using dummy texture): Couldn't open '/res/industrial/gfx-enemy/duker/duker-end.deprecated.png': not found
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:84 Image '/res/industrial/gfx-enemy/duker/duker-head.png' doesn't exist. Attempting to load ".deprecated" version.
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:439 Couldn't load texture '/res/industrial/gfx-enemy/duker/duker-head.png' (now using dummy texture): Couldn't open '/res/industrial/gfx-enemy/duker/duker-head.deprecated.png': not found
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:84 Image '/res/industrial/gfx-enemy/duker/duker-head.png' doesn't exist. Attempting to load ".deprecated" version.
[WARNING] /home/tylerandari12/STcompilation/supertux/src/video/texture_manager.cpp:376 Couldn't load texture '/res/industrial/gfx-enemy/duker/duker-head.png' (now using dummy texture): Couldn't open '/res/industrial/gfx-enemy/duker/duker-head.deprecated.png': not found
[FATAL] /home/tylerandari12/STcompilation/supertux/src/supertux/main.cpp:745 Unexpected exception: Cannot load image.
```
(The game closed after this was printed.)

`duker-body.png`, `duker-end.png`, and `duker-head.png` are the three images I removed so i could test. `duker-head.png` was the only image on a scripted object. Im not sure why the other two also tried to load.

I found another thing, which Im not sure if its related directly to this PR, but I might as well mention it. When this happened I thought "Well that sucks, but if I open the game again I should be able to add the images back in after opening and hit reload resources to make them appear.", but when I opened the level again, the scripted object which had the image was not there at all. I closed the game, added the images back in, and reopened the level, and the scripted object was back again.

Vankata453 commented:
> `duker-body.png`, `duker-end.png`, and `duker-head.png` are the three images I removed so i could test. `duker-head.png` was the only image on a scripted object. Im not sure why the other two also tried to load.

It reloads everything.

> I found another thing, which Im not sure if its related directly to this PR, but I might as well mention it. When this happened I thought "Well that sucks, but if I open the game again I should be able to add the images back in after opening and hit reload resources to make them appear.", but when I opened the level again, the scripted object which had the image was not there at all. I closed the game, added the images back in, and reopened the level, and the scripted object was back again.

Doesn't matter where the script object is, the resource is loaded into the game separately in its own manager.

MatusGuy commented on src/supertux/menu/debug_menu.cpp:
```diff
@@ -71,6 +75,15 @@ DebugMenu::DebugMenu() :
+      SpriteManager::current()->reload();
+      TileManager::current()->reload();
+    })
+    .set_help(_("Reloads all fonts, textures, sprites and tilesets."));
```
The fonts don't reload.

Vankata453 replied:
`Resources::current()->reload()`

MatusGuy reviewed (approved):
Fonts don't reload, but because most modders don't care about changing the fonts, this should be good to go.

Vankata453 commented:
It's not, because it's supposed to do that too. Will look into it.

Vankata453 commented:
TTF fonts now properly reload.
